src/sqlfluff/dialects/dialect_vertica.py

```diff
@@ -0,0 +1,2036 @@
+                        Ref.keyword("ESCAPE"),
+                        Ref("Tail_Recurse_Expression_A_Grammar"),
+                        optional=True,
+                    ),
+                ),
+                Sequence(
+                    Ref("BinaryOperatorGrammar"),
+                    Ref("Tail_Recurse_Expression_A_Grammar"),
+                ),
+                Ref("InOperatorGrammar"),
+                Sequence(
+                    "IS",
+                    Ref.keyword("NOT", optional=True),
+                    Ref("IsClauseGrammar"),
+                ),
+                Ref("IsNullGrammar"),
+                Sequence(
+                    Sequence(
+                        Ref("InterpolateGrammar"),
+                    ),
+                    Ref("Expression_A_Grammar"),
+                ),
+                Ref("NotNullGrammar"),
+                Ref("CollateGrammar"),
+                Sequence(
+                    Ref.keyword("NOT", optional=True),
+                    "BETWEEN",
+                    Ref("Expression_B_Grammar"),
+                    "AND",
+                    Ref("Tail_Recurse_Expression_A_Grammar"),
+                ),
+            )
+        ),
+    ),
+    ComparisonOperatorGrammar=OneOf(
+        Ref("EqualsSegment"),
+        Ref("NullEqualsSegment"),
+        Ref("GreaterThanSegment"),
+        Ref("LessThanSegment"),
+        Ref("GreaterThanOrEqualToSegment"),
+        Ref("LessThanOrEqualToSegment"),
+        Ref("NotEqualToSegment"),
+        Ref("LikeOperatorSegment"),
+        Ref("IsDistinctFromGrammar"),
+    ),
+    JoinTypeKeywordsGrammar=OneOf(
+        "ANTI",
+        "SEMIALL",
+        "SEMI",
+        Sequence("NULLAWARE", "ANTI"),
+        "INNER",
+        Sequence(
+            OneOf(
+                "FULL",
+                "LEFT",
+                "RIGHT",
+            ),
+            Ref.keyword("OUTER", optional=True),
+        ),
+        Sequence(
+            "NATURAL",
+            OneOf(
+                "INNER",
+                Sequence(
+                    OneOf("RIGHT", "LEFT", "FULL"),
+                    "OUTER"
+                ),
+            ),
+        ),
+    ),
+    ArithmeticBinaryOperatorGrammar=OneOf(
+        Ref("PlusSegment"),
+        Ref("MinusSegment"),
+        Ref("DivideSegment"),
+        Ref("IntegerDivideSegment"),
+        Ref("MultiplySegment"),
+        Ref("ModuloSegment"),
+        Ref("BitwiseAndSegment"),
+        Ref("BitwiseOrSegment"),
+        Ref("BitwiseXorSegment"),
+        Ref("BitwiseLShiftSegment"),
+        Ref("BitwiseRShiftSegment"),
+    ),
+    # Vertica supports the non-standard ISNULL and NONNULL comparison operators. See
+    # https://docs.vertica.com/latest/en/sql-reference/language-elements/operators/null-operators/
+    IsNullGrammar=Ref.keyword("ISNULL"),
+    NotNullGrammar=Ref.keyword("NOTNULL"),
+)
+
+
+class ShorthandCastSegment(ansi.ShorthandCastSegment):
+    """A casting operation using '::'."""
```

Comment:
Would be good to explain why overriding. I think here it's only to `allow_gaps` is that so?

Reply:
Actually, the main reason is adding an additional operator for casting `::!` (the one which we discussed in slack). I will update the description